eng/pipelines/doc-index.yml

```diff
@@ -0,0 +1,86 @@
+          script: $(Build.BinariesDirectory)/docfx/docfx.exe init -q
+          workingDirectory: $(Build.SourcesDirectory)
+          failOnStderr: true
+
+      - pwsh: New-Item -Path . -Name "templates" -ItemType "directory"
+        displayName: Create Template Directory
+        workingDirectory: $(Build.SourcesDirectory)/docfx_project/
+
+      # - task: CopyFiles@2
+      #   displayName: Copy Template to templates folder
+      #   inputs:
+      #     sourceFolder: $(Build.SourcesDirectory)/eng/tools/generate-static-index/static-files/templates
+      #     content: "**/*"
+      #     targetFolder: $(Build.SourcesDirectory)/docfx_project/templates
+      #     overWrite: true
+
+      - pwsh: ls
+        workingDirectory: $(Build.SourcesDirectory)/docfx_project/
+      - pwsh: ls
+        workingDirectory: $(Build.SourcesDirectory)/docfx_project/templates/
+
+      - pwsh: |
+          Copy-Item "$(Build.SourcesDirectory)/eng/tools/generate-static-index/static-files/docfx.json" -Destination "$(Build.SourcesDirectory)/docfx_project/" -Force
+        displayName: Copy over docfx.json
+
+      - script: |
+          npm install
+        workingDirectory: $(System.DefaultWorkingDirectory)/eng/tools/generate-static-index
+        displayName: "Install tool dependencies"
+
+      - pwsh: |
+          node $(Build.SourcesDirectory)/eng/tools/generate-static-index/index.js
+        displayName: "Generate Index Toc"
+
+      - pwsh: ls
+        workingDirectory: $(Build.SourcesDirectory)/docfx_project/api
+
+      - pwsh: |
+          New-Item -Path "$(Build.SourcesDirectory)/docfx_project" -Name "toc.yml" -Force
+          Add-Content -Path "$(Build.SourcesDirectory)/docfx_project/toc.yml" -Value "- name: Azure SDK for JavaScript APIs`r`n  href: api/`r`n  homepage: api/index.md"
+          Copy-Item "$(Build.SourcesDirectory)/README.md" -Destination "$(Build.SourcesDirectory)/docfx_project/api/index.md" -Force
+          Copy-Item "$(Build.SourcesDirectory)/README.md" -Destination "$(Build.SourcesDirectory)/docfx_project/index.md" -Force
+        displayName: Update toc.yml and index
+
+      - task: CmdLine@2
+        displayName: Build Doc Content
+        inputs:
+          script: $(Build.BinariesDirectory)/docfx/docfx.exe build
+          workingDirectory: $(Build.SourcesDirectory)/docfx_project/
+          failOnStderr: true
+
+      - pwsh: |
+          Copy-Item "$(Build.SourcesDirectory)/eng/tools/generate-static-index/static-files/assets/*" -Destination "$(Build.SourcesDirectory)/docfx_project/_site/" -Force
+          Get-Content "$(Build.SourcesDirectory)/eng/tools/generate-static-index/static-files/main.js" |Out-File "$(Build.SourcesDirectory)/docfx_project/_site/styles/main.js"
+          Get-Content "$(Build.SourcesDirectory)/eng/tools/generate-static-index/static-files/docfx.css" |Out-File "$(Build.SourcesDirectory)/docfx_project/_site/styles/docfx.css"
+        displayName: Replace site assets
+
+      - task: CopyFiles@2
```

Comment:
Why switch to the copyfile tasks here where you are copying in pswh most other places?

Reply:
done